dds/DCPS/DataReaderImpl.cpp

```diff
@@ -3632,6 +3576,49 @@ void DataReaderImpl::OnDataAvailable::execute()
     subscriber->set_status_changed_flag(::DDS::DATA_ON_READERS_STATUS, false);
   }
 }
+void DataReaderImpl::initialize_lookup_maps()
+{
+  // These all start at 1 (0 mask is bogus) and include the full mask (any)
+  for (CORBA::ULong is = 1; is <= MAX_SAMPLE_STATE_MASK; ++is) {
+    for (CORBA::ULong iv = 1; iv <= MAX_VIEW_STATE_MASK; ++iv) {
+      for (CORBA::ULong ii = 1; ii <= MAX_INSTANCE_STATE_MASK; ++ii) {
+        combined_state_lookup_[to_combined_states(is, iv, ii)] = HandleSet();
+      }
+    }
+  }
```

Comment:
```suggestion
  for (CORBA::ULong is = 1; is <= MAX_SAMPLE_STATE_MASK; ++is) {
    for (CORBA::ULong iv = 1; iv <= MAX_VIEW_STATE_MASK; iv <<= 1) {
      for (CORBA::ULong ii = 1; ii <= MAX_INSTANCE_STATE_MASK; <<= 1) {
        combined_state_lookup_[to_combined_states(is, iv, ii)] = HandleSet();
      }
    }
  }
```

Reply:
I don't think this will work. My understanding is that the combined lookup states need to cover any valid mask combinations a user can throw at us. We can disregard anything with 0 sample state, 0 view state, or 0 instance state, since those are invalid requests, but that leaves 63 (3 * 3 * 7) lists + 1 empty list (for invalid / bogus requests, currently mapped to combined state of '0') that get maintained in this map.

Reply:
Yeah.  This is bogus.